UI/common-settings.cpp

```diff
@@ -0,0 +1,183 @@
+#include "common-settings.hpp"
+
+#include "audio-encoders.hpp"
+
+bool IsAdvancedMode(config_t *config)
+{
+	const char *outputMode = config_get_string(config, "Output", "Mode");
+	return (strcmp(outputMode, "Advanced") == 0);
+}
+
+OBSData CommonSettings::GetDataFromJsonFile(const char *jsonFile)
+{
+	char fullPath[512];
+	obs_data_t *data = nullptr;
+
+	int ret = GetProfilePath(fullPath, sizeof(fullPath), jsonFile);
+	if (ret > 0) {
+		BPtr<char> jsonData = os_quick_read_utf8_file(fullPath);
+		if (!!jsonData) {
+			data = obs_data_create_from_json(jsonData);
+		}
+	}
+
+	if (!data)
+		data = obs_data_create();
+	OBSData dataRet(data);
+	obs_data_release(data);
```

Comment:
OBSData is a smart pointer to an obs_data_t, right? Would be better to use this instead of explicit releases.

Reply:
this is the pattern used around the entire app and I'm aiming to be consistent.

Reply:
Also looks like `OBSData` is a `OBSRef` defined in template `template<typename T, void addref(T), void release(T)> class OBSRef {` , not a standard c++ shard ptr